### PR TITLE
MessagingSession prefetch supports sort for selecting which channels are prefetched

### PR DESCRIPTION
**Pull request by dpwspoon**
**Issue #:**

**Description of changes:**

Add supports for selecting which channels are prefetched via MessagingSession configuration with new enum PrefetchSortBy

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*

1. Run the messagingSession demo here: https://github.com/aws/amazon-chime-sdk-js/tree/main/demos/browser.  `npm run start --app=messagingSession`
2. connect with various check boxes on prefetch for, 
3. when connect with prefetch last-message-timestamp confirm websocket url (in network tab) sends prefetch-by=last-message-timestamp in url query

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
yes

4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
yes, this adds a sort-by to prefetch, no we have not 100% agreed on this yet, this CR will not be merged until agreed on

5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
yes, this adds a sort-by to prefetch, no we have not 100% agreed on this yet, this CR will not be merged until agreed on

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
yes

### CODE REVIEW

File: demos/browser/app/messagingSession/messagingSession.ts [hunk 1]
```diff
@@ -59,10 +60,15 @@ export class DemoMessagingSessionApp implements MessagingSessionObserver {
...
         this.userArn = (document.getElementById('userArn') as HTMLInputElement).value;
         this.sessionId = (document.getElementById('sessionId') as HTMLInputElement).value;
-        this.prefetchOn = (document.getElementById('prefetchOn') as HTMLInputElement).checked;
+        this.prefetchSortByUnread = (document.getElementById('prefetchSortByUnread') as HTMLInputElement).checked;
+        this.prefetchSortByLastMessageTimestamp = (document.getElementById('prefetchSortByLastMessageTimestamp') as HTMLInputElement).checked;
         this.configuration = new MessagingSessionConfiguration(this.userArn, this.sessionId, endpoint.Endpoint.Url, chime);
-        if (this.prefetchOn) {
+        if (this.prefetchSortByUnread) {
+          this.configuration.prefetchOn = PrefetchOn.Connect;
+          this.configuration.prefetchSortBy = PrefetchSortBy.unread
```
Comment (jingwy) [inline demos/browser/app/messagingSession/messagingSession.ts:68]: Should it be `PrefetchSortBy.Unread`?

File: demos/browser/app/messagingSession/messagingSession.html [hunk 2]
```diff
@@ -24,8 +24,12 @@ <h1 class="h3 mb-3 font-weight-normal">Messaging Session</h1>
...
         </div>
         <div class="form-check" style="text-align: left;">
-          <input type="checkbox" id="prefetchOn" class="form-check-input">
-          <label for="prefetchOn" class="form-check-label">Connect with prefetch on</label>
+          <input type="radio" id="prefetchOff" name="prefetchSettings" value="off" checked>
+          <label for="prefetchOff">prefetch off</label><br>
+          <input type="radio" id="prefetchSortByUnread" name="prefetchSettings" value="unread">
+          <label for="prefetchSortByUnread">prefetch sorted by unread</label><br>
+          <input type="radio" id="prefetchSortByLastMessageTimestamp" name="prefetchSettings" value="last-message-timestamp">
+          <label for="prefetchSortByLastMessageTimestamp">prefetch sorted by  last message timestamp</label><br>
```
Comment (jingwy) [inline demos/browser/app/messagingSession/messagingSession.html:32]: nit: Extra space.

File: docs/classes/ApplicationMetadata.html [hunk 3]
```diff
@@ -55,7 +55,7 @@
 					<a href="../modules.html">amazon-chime-sdk-js</a>
 				</li>
 				<li>
-					<a href="applicationmetadata.html">ApplicationMetadata</a>
+					<a href="ApplicationMetadata.html">ApplicationMetadata</a>
```
Comment (jingwy) [inline docs/classes/ApplicationMetadata.html:58]: Why do those doc files change? Are those expected?

Comment (dpwspoon) [reply docs/classes/ApplicationMetadata.html:58]: I don't know but think so, the `npm run build:release` on updated them which is the directions in the PR

File: demos/browser/app/messagingSession/messagingSession.ts (hunk 1 above)
Comment (dpwspoon) [reply demos/browser/app/messagingSession/messagingSession.ts:68]: thanks, will fix!

File: demos/browser/app/messagingSession/messagingSession.html (hunk 2 above)
Comment (dpwspoon) [reply demos/browser/app/messagingSession/messagingSession.html:32]: thanks, will fix!

**Review by ltrung — APPROVED** [review]

**Review by jingwy — APPROVED** [review]

**Pull request merged**